latitudesh/terraform-provider-latitudesh
# Provider server lock feature

#### What does this PR do?
Implements server lock and unlock, add deprecation warning to `allow_reinstall`

#### Description of Task to be completed?
- Add locked attribute to servers
- implement both lock and unlocking when updating a server
- Add a deprecation warning to allow_reinstall informing users about the locked attribute

#### How should this be manually tested?
You should run the automated test suite to ensure that the server resource still works fine:

```
export LATITUDESH_AUTH_TOKEN=<YOUR-API-TOKEN> 
TF_ACC=1 go test ./... -run "TestAccServer" -v
```
also, you can follow [Notion - Developing the Terraform Provider Locally](https://www.notion.so/Developing-the-Terraform-Provider-Locally-6682fe3df39d4003b519301e630f51ef?pvs=27&qid=1%3A8c80df71-563c-4c6a-905d-be4a225ec805%3A0) to test the provider locally. Don't forget to change the server locked attribute and verify how this impacts updates and deletions.

#### Any background context you want to provide?

#### What are the relevant GitHub issues (if any)?

#### Screenshots (if appropriate):

#### Questions:
